js/modules/k6/http/error_codes.go

```diff
@@ -0,0 +1,137 @@
+	"golang.org/x/net/http2"
+)
+
+type errCode uint32
+
+const (
+	// non specific
+	defaultErrorCode          errCode = 1000
+	defaultNetNonTCPErrorCode errCode = 1010
+	// DNS errors
+	defaultDNSErrorCode    errCode = 1100
+	dnsNoSuchHostErrorCode errCode = 1101
+	blackListedIPErrorCode errCode = 1110
+	// tcp errors
+	defaultTCPErrorCode      errCode = 1200
+	tcpBrokenPipeErrorCode   errCode = 1201
+	netUnknownErrnoErrorCode errCode = 1202
+	tcpDialErrorCode         errCode = 1210
+	tcpDialTimeoutErrorCode  errCode = 1211
+	tcpDialRefusedErrorCode  errCode = 1212
+	tcpResetByPeerErrorCode  errCode = 1220
+	// TLS errors
+	defaultTLSErrorCode           errCode = 1300
+	x509UnknownAuthorityErrorCode errCode = 1310
+	x509HostnameErrorCode         errCode = 1311
+
+	// HTTP2 errors
+	// defaultHTTP2ErrorCode errCode = 1600 // commented because of golint
+	// HTTP2 GoAway errors
+	unknownHTTP2GoAwayErrorCode errCode = 1610
+	// errors till 1611 + 13 are other HTTP2 GoAway errors with a specific errCode
+
+	// HTTP2 Stream errors
+	unknownHTTP2StreamErrorCode errCode = 1630
+	// errors till 1631 + 13 are other HTTP2 Stream errors with a specific errCode
+
+	// HTTP2 Connection errors
+	unknownHTTP2ConnectionErrorCode errCode = 1650
+	// errors till 1651 + 13 are other HTTP2 Connection errors with a specific errCode
+)
+
+var (
+	tcpResetByPeerErrorCodeMsg  = "write: connection reset by peer"
+	tcpDialTimeoutErrorCodeMsg  = "dial: i/o timeout"
+	tcpDialRefusedErrorCodeMsg  = "dial: connection refused"
+	tcpBrokenPipeErrorCodeMsg   = "write: broken pipe"
+	netUnknownErrnoErrorCodeMsg = "%s: unknown errno `%d` on %s with message `%s`"
+	dnsNoSuchHostErrorCodeMsg   = "lookup: no such host"
+	blackListedIPErrorCodeMsg   = "ip is blacklisted"
+	http2GoAwayErrorCodeMsg     = "http2: received GoAway with http2 ErrCode %s"
+	http2StreamErrorCodeMsg     = "http2: stream error with http2 ErrCode %s"
+	http2ConnectionErrorCodeMsg = "http2: connection error with http2 ErrCode %s"
+	x509HostnameErrorCodeMsg    = "x509: certificate doesn't match hostname"
+)
+
+func http2ErrCodeOffset(code http2.ErrCode) errCode {
+	if code > http2.ErrCodeHTTP11Required {
+		return 0
+	}
+	return 1 + errCode(code)
+}
+
+// returns the errorCode and a specific error message for given error. If errror message is empty
+// than the original error string should be used
+func errorCodeForError(err error) (errCode, string) {
+	switch e := errors.Cause(err).(type) {
+	case *net.DNSError:
+		switch e.Err {
+		case "no such host": // defined as private in the go stdlib
+			return dnsNoSuchHostErrorCode, dnsNoSuchHostErrorCodeMsg
+		default:
+			return defaultDNSErrorCode, ""
+		}
+	case netext.BlackListedIPError:
+		return blackListedIPErrorCode, blackListedIPErrorCodeMsg
+	case *http2.GoAwayError:
+		// TODO: Add different error for all errcode for goaway
+		return unknownHTTP2GoAwayErrorCode + http2ErrCodeOffset(e.ErrCode), fmt.Sprintf(http2GoAwayErrorCodeMsg, e.ErrCode)
```

Comment:
Not sure if `unknown` is the best word choice here and below, since we likely have the specific error codes and know the actual errors. `base` or something like that seems better, i.e. `baseHTTP2GoAwayErrorCode`, `baseHTTP2StreamErrorCode`, etc.

Reply:
Yeah but this specific one will be used in case of unknown [http2 ErrCode](https://godoc.org/golang.org/x/net/http2#ErrCode) which seems unlikely but I decided to be confidence instead of getting error 1721 and than having to figure out it is actually a GoAway with some strange ErrCode. I also don't like the name though ... maybe default is fine after alll and just the other ones are specifics ?